hibernate-core/src/main/java/org/hibernate/tuple/entity/BytecodeEnhancementMetadataPojoImpl.java

```diff
@@ -140,8 +141,14 @@ public PersistentAttributeInterceptable createEnhancedProxy(EntityKey entityKey,
 		final PersistenceContext persistenceContext = session.getPersistenceContext();
 
 		// first, instantiate the entity instance to use as the proxy
-		final PersistentAttributeInterceptable entity = (PersistentAttributeInterceptable) persister.getEntityTuplizer().instantiate( identifier, session );
+		final EntityTuplizer entityTuplizer = persister.getEntityTuplizer();
+		final PersistentAttributeInterceptable entity = (PersistentAttributeInterceptable) entityTuplizer
+				.instantiate( identifier, session );
 
+		// clear the fields that are marked as dirty in the dirtyness tracker
+		if ( entity instanceof SelfDirtinessTracker ) {
+			( (SelfDirtinessTracker) entity ).$$_hibernate_clearDirtyAttributes();
```

Comment:
But didn't you just instantiate the entity?

Reply:
what I noticed is that when  
```
@Entity(name = "Client")
public static class Client {
        ....
	@Embedded
	private Log log = new Log();
}
```
is instantiated, because the field `log` has an assigned initial value `new Log()` a call of `$$_hibernate_write_log` and then of `$$_hibernate_trackChange` is triggered so the newly instantiated entity results dirty.

Reply:
probably a better approach is to add  to the enhanced class the ability to reset the tracker on exit of the constructor method

Reply:
No, you explained it.. that's reasonable

Reply:
An alternative (same net effect) would be to disable dirty tracking during construction